src/shmem_comm.h

```diff
@@ -336,6 +336,7 @@ shmem_internal_atomicv(shmem_ctx_t ctx, void *target, const void *source,
         shmem_internal_fetch_atomic(ctx, ((uint8_t *) target) + (i * type_size),
                                     ((uint8_t *) source) + (i * type_size), &tmp_fetch, type_size,
                                     pe, op, datatype);
+        shmem_transport_get_wait((shmem_transport_ctx_t *)ctx);
```

Comment:
Can we move this outside the loop over `count`?  I don't quite see why we need to wait after every atomic?

Reply:
Nope, we cant as the loop itself can go beyond the number of atomics that causes the issue on CXI.